### PR TITLE
Fix tools containing nulls

### LINKED ISSUE #16751: Tool containing a null
I killed a zombie and it dropped a curved needle that had "seeing this is a bug" below its own description.

I saved the game and checked the save to see what's going on and in the save, the needle had a null item (item with id "null") as its contents. This didn't change when I unloaded the needle.

That may be the cause of unusable tools described on forums. Tools with contents will not be used as components and possibly as tools either (cans with contents can't be used to boil water).

Tagging as magazine because it is clearly associated with changes to magazine system.

### PR DESCRIPTION
**Pull request by Coolthulhu**
Fixes #16751

### CODE REVIEW

File: src/item_group.cpp [hunk 1]
```diff
@@ -345,7 +345,7 @@ Item_spawn_data::ItemList Item_group::create(int birthday, RecursionList &rec) c
 
     for( auto& e : result ) {
         if( e.is_tool() || e.is_gun() || e.is_magazine() ) {
-            bool spawn_ammo = rng( 0, 99 ) < with_ammo && e.ammo_remaining() == 0;
+            bool spawn_ammo = rng( 0, 99 ) < with_ammo && e.ammo_remaining() == 0 && e.ammo_type() != "NULL";
```
Comment (mugling) [inline src/item_group.cpp]: I think there are more occurrences of this idiom? Possibly the code for spawning items in vehicles is similarly affected?

Comment (Coolthulhu) [inline src/item_group.cpp]: Yeah, looks like it is repeated in vehicle code

**Comment by Coolthulhu** [comment]
Searched for `with_ammo` and `with_mag` and didn't find anything else.

**Comment by mugling** [comment]
Good work on tracking that down. That fix should work correctly but charged tools without a specific ammo type do exist (eg. spraycan).

It might be worth moving that fix upstream within `ammo_set` so that the `is_magazine()` branch refuses to emplace `NULL` ammo immediately after the call to `ammo_unset`?

File: src/item_group.cpp [hunk 2]
```diff
@@ -345,7 +345,7 @@ Item_spawn_data::ItemList Item_group::create(int birthday, RecursionList &rec) c
...
             bool spawn_mag  = rng( 0, 99 ) < with_magazine && !e.magazine_integral() && !e.magazine_current();
 
             if( spawn_mag || spawn_ammo ) {
```
Comment (mugling) [inline src/item_group.cpp]: This looks to be the more suspicious line. An item using an integral magazine doesn't have a default magazine...

**Pull request closed without merging**